include/tlapack/lapack/gemmtr.hpp

```diff
@@ -0,0 +1,319 @@
+/// @file gemmtr.hpp
+/// @author Luis Carlos Gutierrez, Kyle D. Cunningham, and Henricus Bouwmeester
+//
+// Copyright (c) 2025, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef TLAPACK_GEMMTR_HH
+#define TLAPACK_GEMMTR_HH
+
+#include "tlapack/base/utils.hpp"
+
+namespace tlapack {
+
+/**
+ * General triangular matrix-matrix multiply:
+ * \[
+ *     C := \alpha op(A) \times op(B) + \beta C,
+ * \]
+ * where $op(X)$ is one of
+ *     $op(X) = X$,or
+ *     $op(X) = X^H$,
+ * alpha and beta are scalars, and A, B, and C are matrices, with
+ * $op(A)$ an n-by-k matrix, $op(B)$ a k-by-n matrix, and C an n-by-n matrix.
+ *
+ * @param[in] uplo
+ *     What part of the matrix A is referenced,
+ *     the opposite triangle being assumed to be zero:
+ *     - Uplo::Lower: A is lower triangular.
+ *     - Uplo::Upper: A is upper triangular.
+ *     - Uplo::General is illegal (see gemm() instead).
+ *
+ * @param[in] transA
+ *     The operation $op(A)$ to be used:
+ *     - Op::NoTrans:   $op(A) = A$.
+ *     - Op::Trans:     $op(A) = A^T$.
+ *     - Op::ConjTrans: $op(A) = A^H$.
+ *
+ * @param[in] transB
+ *     The operation $op(B)$ to be used:
+ *     - Op::NoTrans:   $op(B) = B$.
+ *     - Op::Trans:     $op(B) = B^T$.
+ *     - Op::ConjTrans: $op(B) = B^H$.
+ *
+ * @param[in] alpha Scalar.
+ * @param[in] A $op(A)$ is an n-by-k matrix.
+ * @param[in] B $op(B)$ is an k-by-n matrix.
+ * @param[in] beta Scalar.
+ * @param[in,out] C is an n-by-n matrix.
+ *
+ * @ingroup blas3
+ */
+template <TLAPACK_MATRIX matrixA_t,
+          TLAPACK_MATRIX matrixB_t,
+          TLAPACK_MATRIX matrixC_t,
+          TLAPACK_SCALAR alpha_t,
+          TLAPACK_SCALAR beta_t,
+          class T = type_t<matrixC_t>,
+          disable_if_allow_optblas_t<pair<matrixA_t, T>,
+                                     pair<matrixB_t, T>,
+                                     pair<matrixC_t, T>,
+                                     pair<alpha_t, T>,
+                                     pair<beta_t, T> > = 0>
```

Comment:
You can safely remove those lines. They only make sense for routines that are implemented by BLAS++ or LAPACK++. All tests should work without them